pm/recipient.go

```diff
@@ -134,6 +136,11 @@ func (r *recipient) Stop() {
 	close(r.quit)
 }
 
+// ValidateSender checks whether sender is a valid ticket sender
+func (r *recipient) ValidateSender(sender ethcommon.Address) error {
```

Comment:
What I don't like about that is that we would be validating the sender for each ticket in the case where a payment constitutes multiple tickets. This seems like an inefficiency to me as we want to validate the sender of a payment as a whole rather than separate tickets. 

We could return an unacceptable payment error from `ReceiveTicket` but this will still try to redeem winning tickets, which I don't see as fair towards B as these errors can be ephemeral and we don't add credit but are redeeming tickets. 

Then again we could do `if won && !unacceptableReceiveErr {...}` or break the loop upon an `unacceptableReceiveError`

Reply:
Fixed in d4b2a96, but I force pushed since the implementation is substantially different and the diffs wouldn't really matter here I guess.

Reply:
> What I don't like about that is that we would be validating the sender for each ticket in the case where a payment constitutes multiple tickets.

The first ticket should fail validation and we would not validate the rest of the tickets in the batch which seems ok.

> We could return an unacceptable payment error from ReceiveTicket but this will still try to redeem winning tickets

The error returned would already be considered an unacceptable payment error since we are not returning an `AcceptableError` from `ReceiveTicket()` when the sender validation fails.

> which I don't see as fair towards B as these errors can be ephemeral and we don't add credit but are redeeming tickets.

O should use the strategy that best serves its interest - in this case, it should always redeem a valid winning ticket if it receives one. We're treating a sender validation error the same way that we would treat an unacceptable payment error that is triggered due to a sender exceeding its max error count - we will not credit the sender and will not transcode the segment, but if the ticket wins we will still redeem it.